plugins/outputs/sensu/sensu.go

```diff
@@ -0,0 +1,468 @@
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("User-Agent", internal.ProductToken())
+
+	req.Header.Set("Content-Type", defaultContentType)
+	if s.ContentEncoding == "gzip" {
+		req.Header.Set("Content-Encoding", "gzip")
+	}
+
+	if s.ApiKey != nil {
+		bearerType := "Bearer"
+		if s.BackendApiUrl != nil {
+			bearerType = "Key"
+		}
+		req.Header.Set("Authorization", bearerType+" "+*s.ApiKey)
+	}
+
+	for k, v := range s.Headers {
+		if strings.ToLower(k) == "host" {
+			req.Host = v
+		}
+		req.Header.Set(k, v)
+	}
+
+	resp, err := s.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	_, err = ioutil.ReadAll(resp.Body)
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return fmt.Errorf("when writing to [%s] received status code: %d", endpointUrl, resp.StatusCode)
+	}
+
+	return nil
+}
+
+// Resolves the event write endpoint
+func (s *Sensu) GetEndpointUrl() (string, error) {
+	var endpointUrl string
+
+	if s.BackendApiUrl != nil {
+		endpointUrl = *s.BackendApiUrl
+	} else if s.AgentApiUrl != nil {
+		endpointUrl = *s.AgentApiUrl
+	}
+
+	u, err := url.Parse(endpointUrl)
+	if err != nil {
+		return endpointUrl, err
+	}
+
+	if s.BackendApiUrl != nil {
+		namespace := "default"
+		if s.Entity != nil && s.Entity.Namespace != nil {
+			namespace = *s.Entity.Namespace
+		}
+		u.Path = path.Join(u.Path, "/api/core/v2/namespaces", namespace, "events")
+	} else {
+		u.Path = path.Join(u.Path, "/events")
+	}
+	endpointUrl = u.String()
+
+	return endpointUrl, nil
+}
+
+func init() {
+	outputs.Add("sensu-go", func() telegraf.Output {
+		// Default configuration values
+		defaultUrl := defaultUrl
+
+		return &Sensu{
+			AgentApiUrl: &defaultUrl,
+			Timeout:     internal.Duration{Duration: defaultClientTimeout},
+		}
+	})
+}
+
+type OutputEvent struct {
+	Entity    *OutputEntity  `json:"entity,omitempty"`
+	Check     *OutputCheck   `json:"check"`
+	Metrics   *OutputMetrics `json:"metrics"`
+	Timestamp int64          `json:"timestamp"`
+}
+
+func (s *Sensu) Wrapper(metricPoints []*OutputMetric) ([]byte, error) {
+	timestamp := time.Now().Unix()
+
+	var err error
+	entity, err := s.GetEntity()
+	if err != nil {
+		return []byte{}, err
+	}
+
+	check, err := s.GetCheck(metricPoints)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	output, err := json.Marshal(&OutputEvent{
+		Entity: entity,
+		Check:  check,
+		Metrics: &OutputMetrics{
+			Handlers: s.GetHandlers(),
+			Metrics:  metricPoints,
+		},
+		Timestamp: timestamp,
+	})
+
+	return output, err
+}
+
+// Constructs the entity payload
+// Throws when no entity name is provided and fails resolve to hostname
+func (s *Sensu) GetEntity() (*OutputEntity, error) {
+	if s.BackendApiUrl != nil {
+		var entityName string
+		if s.Entity != nil && s.Entity.Name != nil {
+			entityName = *s.Entity.Name
+		} else {
+			defaultHostname, err := os.Hostname()
+
+			if err != nil {
+				return &OutputEntity{}, fmt.Errorf("when resolving hostname")
```

Comment:
Carrying the error upwards by
```suggestion
				return &OutputEntity{}, fmt.Errorf("resolving hostname failed: %v", err)
```
would be nice.